google/cloud/pubsub/internal/tracing_message_batch.cc

```diff
@@ -68,25 +87,27 @@ void TracingMessageBatch::Flush() {
               AttributesList{{"messaging.pubsub.message.link", i++}});
         });
   }
-  auto batch_sink_span_parent =
-      internal::MakeSpan("BatchSink::AsyncPublish",
-                         /*attributes=*/
-                         {{"messaging.pubsub.num_messages_in_batch",
-                           static_cast<std::int64_t>(batch_size)}},
-                         /*links*/ links);
-
+  {
+    std::lock_guard<std::mutex> lk(mu_);
+    batch_sink_parent_span_ =
```

Comment:
Typically, if this needs a mutex then `batch_sink_spans_.push_back()` in line 16 probably does too, and so does the `std::transform()`.  Any reason why they don't?

Reply:
My initial thinking was I should put a mutex around all writes to any member variables.

Since transform is modifying `links` I thought it would be ok. But your comment makes me think it is not... so I added a lock around the transform.